src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -594,7 +594,7 @@ bool value_set_dereferencet::memory_model_bytes(
 
     // possibly need to convert
     if(!base_type_eq(result.type(), to_type, ns))
-      result.make_typecast(to_type);
+      result = typecast_exprt(result, to_type);
   }
```

Comment:
Same here

Reply:
Done (killing this instance of `base_type_eq`)